src/components/ComplexView.test.tsx

```diff
@@ -0,0 +1,13 @@
+import Enzyme, {shallow} from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import ComplexView from "./ComplexView";
+import React, {ReactElement} from "react";
+import {jsJsx} from "ts-loader/dist/constants";
+
+Enzyme.configure({adapter: new Adapter()});
+
+test("When user inputs price and submits then callback is executed and view is changed", () => {
```

Comment:
For this test I would create a component with onClick, which would execute the callback.
And then I would simulate if this component is visible -> then simulate click -> then see if second component is visible.

However in this test you would have to do deep rendering.